### ISSUE TITLE
Permissions to Connect to Hub/Mint

### ISSUE THREAD

**Issue by rigelrozanski** [cosmos/cosmos-economics#30]
@ebuchman @jaekwon @sunnya97 @adrianbrink 

We've talked about this some time ago it would be great to have the logic documented as to how or why permission is required from governance to:
 1. add an existing blockchain with an independant validator set
 2. add a new blockchain which will re-use cosmos validator set 
 3. send zone-specific tokens minted by that blockchain to through the hub

Also (4.) spam protection mechanisms for arbitrary tokens coming from cosmos-hub

Maybe let's link any old issues to this one?

**Comment by rigelrozanski** [comment]
Slack comments:

**adrian** 
It’s too late now :slightly_smiling_face:
Mh, rige and I just spoke about it and maybe we don’t need permissions.
Without permissions, all tokens are prefixed by zone. So photons would be `hub-pth` and atoms would be `hub-atm`, pegged ether would be `peg-eth`. There could be multiple competing peg zones and then we would have `peg1-eth` and `peg2-eth`. Application, zone and UI developers just need to understand that it really matters where a token comes from.

**adrian**
The problem with permissioned zones is that Ethermint can create new tokens in a permissionless way (and that’s the beauty).
With permissioned zones the question is going to be whether every single ERC20 has to be accepted by governance. I think permissioned zones would greatly screw with the beauty of tokens.
jb 
does a new zone need permission to use the validator pool from the main hub? (edited)

**adrian**
yes, shared security will most likely need permissions

**bucky**
the validators have to agree to run the zones …

**Comment by rigelrozanski** [comment]
well - that makes sense - (kinda) - it would also be nice if the validators could run permissionlessly run zones - but nonetheless, other zones with custom validator sets should probably be able to permissionlessly connect to the hub and arbitrarily mint their tokens (prefixed with their zone space) and move them across the hub - the only reason that you’d want to permission every single token is for spam protection against decoy tokens (like `rige/eth`). - I think it’s kind of arbitrary for if governance needs to vote for the connectivity of every single zone to the hub, I’d imagine it may really gum up the jaws of the hub - but who knows if 2 weeks waiting period with an active governance group maybe it’s actually okay? @bucky further thoughts on this?

I guess ultimately if you think about the Ethereum network, anyone is able to create arbitrary ERC20 tokens and have them magically operating without a permissioned vote or uptake from the network, <- how are we to deal with these tokens from the permissioned Ethermint Zone? Does it have special permissions to add ERC20 tokens to the hub… and if so we still need some semblance of spam protections when folks inevitably introduce ERC21 tokens on ethermint named `BTC`, which could be prefixing the zone name so now called `emint/btc` <- moreover if ethermint is going to have special permissions to add ERC21 tokens without special permissions through governance then it seems arbitrary that independently validatated zones wouldn’t have those same permissions.

**Comment by rigelrozanski** [comment]
**jaekwon** 
Only 2 would be permissioned.

**jaekwon**
there are designs that could work more permissionlessly, like just hard forking the atom set, but maybe we can adopt our cooling period protocol to enable permissionless running of chains, where if after a period more than say 67% of the validators participate in that chain, then after the period it is considered adopted by the validators set, and the inactive ones have some grace period to come back online before they are considered faulty by that chain, and IBCd back to the hub to unbond it everywhere. (edited)
But I think first step should maybe just be governance to coordinate.  After a few of those we could try automating it.
But yeah otherwise the fees are permissioned too but nothing else is.
I think we restrict coin names to those that are longer than 5 chars for now.

**rige**
Yeah, makes sense - coin names 5 char seems reasonable ( 26^5 = 11,881,376 unique coins) but should probably still preface the chain which minted on as a coin field

**Comment by rigelrozanski** [comment]
The sentiment that i’m reading is we are moving towards being as permissionless as possible, however if your chain is hoping on piggybacking on the validator set of the hub - anything that runs on a validator set is trusted code to that validator set I believe which means that any new code running on an existing validator set must be approved by those validators (or else it would be an easy attack surface). <- this is of course not the case with something like the EVM, and also not the case if you’re putting up your own validator set to run your blockchain

**Comment by adrianbrink** [comment]
The reason for prefixing with the zone name is that otherwise the first pegzone will just register all ERC20 for eternity or every new project has to fight to keep the correct name. Different zones have different security properties and those should be reflected in the token names.

**Comment by adrianbrink** [comment]
@rigelrozanski @jaekwon I agree. Everything that reuses the hub validator set should go through governance or the above design for semi-automatic adoption.
